src/couch/src/couch_bt_engine.erl

```diff
@@ -759,6 +777,45 @@ set_default_security_object(Fd, Header, Compression, Options) ->
     end.
 
 
+% This function is to rollback pf clustered purge.
+% It replaces purge_tree_state and upurge_tree_state with
+% with purged_docs disk pointer that stores only the last purge request
```

Comment:
This function is wrong as its not a mirror of what happens when we upgrade the state turning the whole purge docs into multiple rows. It also occurs to me that we transform this in such a way that we'd not be able to know how to undo the upgrade since we would not be able to distinguish between an upgrade and purges applied after the upgrade.

Also if we rollback this approach would also likely invalidate all secondary indices since we're grabbing the last purge sequence as well.